python/ml4ir/applications/ranking/tests/test_lr_schedule.py

```diff
@@ -0,0 +1,217 @@
+            initial_learning_rate=0.1,
+            decay_steps=2,
+            decay_rate=0.9,
+        )
+        scheduler = [lrs(i) for i in range(25)]
+        self.assertTrue(self.compare_lr_values(scheduler, gold))
+
+    def test_constant_lr(self):
+        """Test a constant learning rate"""
+        lrs = ExponentialDecay(
+            initial_learning_rate=0.1,
+            decay_steps=10000000,
+            decay_rate=1.0,
+        )
+        for i in range(25):
+            assert round(float(lrs(i).numpy()), 5) == 0.1
+
+    def test_cyclic_lr_in_training_pipeline(self):
+        """Test a cyclic learning rate in model training"""
+        Logger = logging_utils.setup_logging(
+            reset=True,
+            file_name=os.path.join(INPUT_DIR + 'ranklib_test_data', "output_log.csv"),
+            log_to_file=True,
+        )
+
+        io = LocalIO()
+        feature_config = self.parse_config(TFRecordTypeKey.SEQUENCE_EXAMPLE, self.feature_config_yaml_convert_to_clicks,
+                                           io)
+
+        dataset = RelevanceDataset(
+            data_dir=INPUT_DIR + '/ranklib_test_data',
+            data_format=DataFormatKey.RANKLIB,
+            feature_config=feature_config,
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            batch_size=2,
+            file_io=io,
+            preprocessing_keys_to_fns={},
+            logger=Logger,
+            keep_additional_info=KEEP_ADDITIONAL_INFO,
+            non_zero_features_only=NON_ZERO_FEATURES_ONLY,
+            max_sequence_size=319,
+        )
+
+        # Define interaction model
+        interaction_model: InteractionModel = UnivariateInteractionModel(
+            feature_config=feature_config,
+            feature_layer_keys_to_fns={},
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            max_sequence_size=319,
+            file_io=io,
+        )
+
+        # Define loss object from loss key
+        loss: RelevanceLossBase = loss_factory.get_loss(
+            loss_key=LossKey.RANK_ONE_LISTNET, scoring_type=ScoringTypeKey.POINTWISE
+        )
+
+        # Define scorer
+        scorer: ScorerBase = RelevanceScorer.from_model_config_file(
+            model_config_file=self.model_config_file,
+            interaction_model=interaction_model,
+            loss=loss,
+            logger=Logger,
+            file_io=io,
+        )
+
+        optimizer: Optimizer = get_optimizer(file_io=io, model_config_file=self.model_config_file)
+
+        # Combine the above to define a RelevanceModel
+        relevance_model: RelevanceModel = RankingModel(
+            feature_config=feature_config,
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            scorer=scorer,
+            optimizer=optimizer,
+            model_file=None,
+            file_io=io,
+            logger=Logger,
+        )
+        callbacks_list = []
+        lr_list = []
+        my_calback_object = lr_callback(lr_list)
```

Comment:
@mohazahran An alternative implementation to consider is have assert statements in your callback itself. Upto you. This looks fine as well. The one change I would recommend though is to fetch the value using a getter method from the callback object after training as opposed to passing a list by reference into the callback like here.